libp2p/protobuf/minprotobuf.nim

```diff
@@ -657,6 +576,17 @@ proc getField*(pb: ProtoBuffer, field: int,
   else:
     err(res.error)
 
+proc getRequiredField*[T](pb: ProtoBuffer, field: int,
+               output: var T): ProtoResult[void] {.inline.} =
```

Comment:
Now that I think about it, it doesn't really make much sense to have a `Result` type and use a variable for output. `Result` already allows you to express both an error condition and a result value when everything went well. 

IMO a `void` result **only** makes sense when nothing is being returned from the function, in this (and similar) case, there is a return value and it should have been passed as part of the result. This resembles C and it's not very idiomatic to Nim, IMO.

Reply:
Good point

I mostly did that to stay consistent with getField, but also because the generic matching don't work solely on result type, so you would have to do
`? getMandatoryField[PeerId](5)`

(PeerId explicitely)
Not sure if it's a good think or a bad thing

Reply:
I think if you do `let peerId: PeerID = getMandatoryField(5)`, the compiler deduces the generic type? Usually Nim is pretty good at inferring this.

Reply:
I don't think type inferring works on the return type, I'll check to be sure

Reply:
Moved PB to a new issue for better follow up: https://github.com/status-im/nim-libp2p/issues/653